### DISCUSSION TITLE
Installing podman version 4.7 or above in Pop!OS Ubuntu flavored 22.04

### DISCUSSION THREAD

**Discussion by lgonzalezsa** [containers/podman#22310] [Q&A] [answered]
I've been with Pop!OS since 22.04, the standard podman version included in Ubuntu 22.04 is 3.X; very old for my use cases.
Time ago, the podman instructions were pointing to `http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_22.04/` as an alternative way to have a more recent version of podman available for Ubuntu LTS users. The podman documentation was updated and no more reference to that source.
The new docs are here https://podman.io/docs/installation#ubuntu 

Not sure if Ubuntu LTS 24.04 is addressing the need to have a version greater than 4.7 available.

As end user, **what are my alternatives to install podman in my OS?**

In addition, the *Docker compatibility mode* to have /var/run/docker.sock available; not sure if the official procedure is to create the symbolic link manually or could be a similar helper like in macOS where the tool helps to configure. No reference in Linux when we need this for example for compose.

Thank you!

**Comment by eriksjolund** [comment]
> Not sure if Ubuntu LTS 24.04 is addressing the need to have a version greater than 4.7 available.

To find out, I did the following

1. Go to https://en.wikipedia.org/wiki/Ubuntu_version_history and check the _code name_ of Ubuntu 24.04
   __result:__ `noble`
2. Go to https://packages.ubuntu.com
3. In the _Keyword_ field, enter `podman`
4. In the _Distribution_ dropdown list, select _noble_
5. Click _search_
    __result:__ A web page is shown that mentions `podman 4.9.3`

(Ubuntu 24.04 has not yet been released so this is preliminary information)

**Reply by lgonzalezsa** [reply]
Good to read this preliminary information, I think the problem will be over time, the release will have near to up to date versions, but as happens today with 22.04, the gap increases, now the latest podman published version is 5.x vs the 3.x shipped by 22.04 LTS. Maybe I am touching a packaging topic instead of a podman related topic, on what alternatives we could have as users of Ubuntu based distros like Pop!OS.

On the other topic: Docker compatibility mode; is it there documentation published for Linux, on how to have `/var/run/docker.sock available`?

**Reply by afbjorklund** [reply]
Most likely, in two years you will be wondering why your 24.04 LTS has 4.x instead of the brand new 6.x in Fedora ?

I don't think you need to create /var/run/docker.sock, if you change your docker configuration to point at `podman.sock`

**Reply by afbjorklund** [reply]
Something like this: `export DOCKER_HOST=unix://$XDG_RUNTIME_DIR/podman/podman.sock`

https://www.docker.com/blog/how-to-deploy-on-remote-docker-hosts-with-docker-compose/

`docker context create podman --docker "host=unix://$XDG_RUNTIME_DIR/podman/podman.sock"`

`docker context use podman`

**Reply by lgonzalezsa** [reply]
Exactly, what are the alternatives to have near to latest version of podman on Ubuntu based OS?
I mean, is good to have the kubic repository because at least am able to use podman 4.6.2 but not sure if similar initiative will be available for 24.04 in the future to at least not be way behind from latest podman versions.

Thanks for the pointer to contexts, will test that for my docker-compose usage.

**Reply by afbjorklund** [reply]
We talked about alternatives, when it comes to how to run Podman on Linux (as in Ubuntu, or Debian):

* https://github.com/containers/podman-desktop/discussions/5762

1) Use a virtual machine, to make the experience at least more similar to the one on Mac and Windows

2) Compile the newer versions from source, including using containers to make a reproducible build

**Reply by afbjorklund** [reply, marked as answer]
> Maybe I am touching a packaging topic instead of a podman related topic
* #17362
* #14302

**Reply by lgonzalezsa** [reply]
Thank you @afbjorklund #14302 helped me find out I can use an alternative approach via homebrew as a third option from your list.

**Reply by afbjorklund** [reply]
Sounds good, but currently it is [mentioned](https://podman.io/docs/installation) as "Though not recommended, Podman can also be obtained through Homebrew, the package manager". so I think it will be on the same support level as the discontinued Kubic repositories... That is: unsupported*

> Since Brew is a community-maintained package manager, we cannot guarantee stability of Brew installs of Podman. Thus, installing via Brew is not recommended.

\* But I think that just means that it is not supported by the Podman team?

i.e. Ubuntu is supported by Ubuntu, and Brew is supported by Brew, etc

**Reply by afbjorklund** [reply]
In the end I think it was just going with some "warning, you are running Ubuntu!"

* https://github.com/containers/podman-desktop/issues/4345

**Comment by lgonzalezsa** [comment]
Confirmed at least Ubuntu daily build has podman 4.9.3
![image](https://github.com/containers/podman/assets/704686/c8f82a93-f51f-4848-a758-e298dfb6e8bd)
